workflow/artifacts/git/git.go

```diff
@@ -39,7 +39,9 @@ func (g *GitArtifactDriver) Load(inputArtifact *wfv1.Artifact, path string) erro
 			_ = os.Remove(tmpfile.Name())
 		}()
 		re := regexp.MustCompile("@(.*):")
-		repoHost := re.FindStringSubmatch(inputArtifact.Git.Repo)[1]
+		incomingHost := re.FindStringSubmatch(inputArtifact.Git.Repo)[1]
+		repoHost := fmt.Sprintf("%s github.com gitlab.com bitbucket.org", incomingHost)
```

Comment:
Is this functionality something that was lost with the move to go-git? Obviously hard-wiring the three main ones will work for most users, but how would this work with people with self hosted git?

Reply:
@jessesuen it was never there, since it wasn't cloning over ssh. We can make it configurable or disable strict host key checking, that way it will be able to clone from anywhere. https://github.com/src-d/go-git/issues/454

Maybe that's a better solution. Let me know I can make the changes